core/arch/arm/kernel/thread_spmc.c

```diff
@@ -326,46 +327,104 @@ static bool is_my_uuid(uint32_t w0, uint32_t w1, uint32_t w2, uint32_t w3)
 	       w2 == 0x02005ebc && w3 == 0x1bc5d5a5;
 }
 
-static void handle_partition_info_get(struct thread_smc_args *args,
-				      struct ffa_rxtx *rxtx)
+void spmc_create_partition_entry(struct ffa_partition_info *fpi,
+				 uint16_t endpoint_id)
+{
+	fpi->id = endpoint_id;
+	fpi->execution_context = CFG_TEE_CORE_NB_CORE;
+	fpi->partition_properties = BIT(0) | BIT(1);
+}
+
+static uint32_t handle_partition_info_get_all(int *rc, struct ffa_rxtx *rxtx)
 {
 	uint32_t ret_fid = 0;
-	int rc = 0;
 
-	if (!is_nil_uuid(args->a1, args->a2, args->a3, args->a4) &&
-	    !is_my_uuid(args->a1, args->a2, args->a3, args->a4)) {
+	if (rxtx->size && rxtx->tx_is_mine) {
+		struct ffa_partition_info *fpi = rxtx->tx;
+
+		spmc_create_partition_entry(fpi, my_endpoint_id);
+		ret_fid = FFA_SUCCESS_32;
+		rxtx->tx_is_mine = false;
+		*rc = 1;
+		fpi++;
+
+		if (IS_ENABLED(CFG_SECURE_PARTITION)) {
+			size_t max_elements = rxtx->size /
+					      sizeof(struct ffa_partition_info)
+					      - 1;
```

Comment:
prefer minus sign at end of the previous line.
Maybe:
```c
			size_t max_elements = (rxtx->size / sizeof(*fpi)) - 1;
```
or
```c
			size_t max_elts = rxtx->size /
					  sizeof(struct ffa_partition_info) - 1;
```

Generic comment: caller of FFA_PARTITION_INFO_GET cannot reach last partitions when extx buffer size is too small to fit them all? How to overcome that?

Reply:
The code could be changed to be a bit less nested. If we instead check for the error case and do an exit early we wouldn't get this deep. It also has the advantage that the may code look a bit less complicated.

Reply:
Done